go/src/koding/klient/machine/machinegroup/machinegroup.go

```diff
@@ -176,3 +247,16 @@ func (g *Group) dynamicAddr(id machine.ID) client.DynamicAddrFunc {
 		return g.address.Latest(id, network)
 	}
 }
+
+// dynamicClient creates dynamic client functor that is used for mount sync
+// connections.
+func (g *Group) dynamicClient() mountsync.DynamicClientFunc {
+	return func(mountID mount.ID) (client.Client, error) {
```

Comment:
No need for closure here. You can just define this method on Group:

```go
func (g *Group) dynamicClient() (Client, error) {
```

And then use method selector:

```go
	syncOpts := mountsync.SyncOpts{
		ClientFunc: g.dynamicClient,
```